src/components/ha-md-dialog.ts

```diff
@@ -0,0 +1,148 @@
+import { MdDialog } from "@material/web/dialog/dialog";
+import { css } from "lit";
+import { customElement, property } from "lit/decorators";
+
+/**
+ * Based on the home assistant design: https://design.home-assistant.io/#components/ha-dialogs
+ *
+ */
+@customElement("ha-md-dialog")
+export class HaMdDialog extends MdDialog {
+  /**
+   * When true the dialog will not close when the user presses the esc key or press out of the dialog.
+   */
+  @property({ attribute: "disable-cancel-action", type: Boolean })
+  public disableCancelAction = false;
+
+  private _dialogPolyfill: any;
+
+  constructor() {
+    super();
+
+    this.addEventListener("cancel", this._handleCancel);
+    this.addEventListener("open", this._handleOpen);
+
+    if (typeof HTMLDialogElement !== "function") {
+      this._dialogPolyfill = import("dialog-polyfill");
+    }
+
+    // if browser doesn't support animate API disable open/close animations
+    if (this.animate === undefined) {
+      this.quick = true;
+    }
+  }
+
+  firstUpdated() {
+    super.firstUpdated();
+
+    // setup polyfill dialog for older browsers
+    if (typeof HTMLDialogElement !== "function") {
+      this._setupDialogPolyfill();
+    }
+  }
+
+  private async _setupDialogPolyfill() {
+    this._loadPolyfillStylesheet("/static/polyfills/dialog-polyfill.css");
+    const dialog = this.shadowRoot?.querySelector(
+      "dialog"
+    ) as HTMLDialogElement;
+    const dialogPolyfill = await this._dialogPolyfill;
+    dialogPolyfill.default.registerDialog(dialog);
+  }
+
+  private async _handleOpen(openEvent: Event) {
+    // prevent open in older browsers and wait for polyfill to load
+    if (typeof HTMLDialogElement !== "function") {
+      openEvent.preventDefault();
+      await this._dialogPolyfill;
```

Comment:
You actually have to await `_setupDialogPolyfill`

Reply:
ok thanks, updated.